core/templates/dev/head/services/UserService.ts

```diff
@@ -17,48 +17,56 @@
  */
 
 oppia.factory('UserService', [
-  '$http', '$q', '$window', 'UrlInterpolationService', 'UserInfoObjectFactory',
-  'DEFAULT_PROFILE_IMAGE_PATH',
-  function($http, $q, $window, UrlInterpolationService, UserInfoObjectFactory,
-      DEFAULT_PROFILE_IMAGE_PATH) {
+  '$http', '$q', '$window', 'UrlInterpolationService',
+  'UserInfoObjectFactory', 'DEFAULT_PROFILE_IMAGE_PATH',
+  function($http, $q, $window, UrlInterpolationService,
+      UserInfoObjectFactory, DEFAULT_PROFILE_IMAGE_PATH) {
     var PREFERENCES_DATA_URL = '/preferenceshandler/data';
 
     var userInfo = null;
 
     var getUserInfoAsync = function() {
-      if (GLOBALS.userIsLoggedIn) {
-        if (userInfo) {
-          return $q.resolve(userInfo);
-        }
-        return $http.get(
-          '/userinfohandler'
-        ).then(function(response) {
-          userInfo = UserInfoObjectFactory.createFromBackendDict(response.data);
-          return userInfo;
-        });
-      } else {
-        return $q.resolve(UserInfoObjectFactory.createDefault());
+      var currentUrl = $window.location.href;
+      if (currentUrl.includes('signup')) {
```

Comment:
How we are going to handle this URL: `https://www.oppia.org/search/find?q=signup&language_code=(%22en%22)`?

Reply:
oh! Thanks for pointing it out. I think we should handle this issue by checking pathname of the url. 
If pathname of the url is '/signup' , we should not make /userinfohandler while user is filling the form.
I think this will work right?

Reply:
```javascript
var currentPath = UrlService.getCurrentUrl().pathname;
      if (currentPath === '/signup') {
```

Reply:
Whatever you go with, make it its own method in URLService, give it a clear conceptual name, and add thorough unit tests for it.